src/widgets/search-box/search-box.js

```diff
@@ -376,161 +303,94 @@ function addDefaultAttributesToInput(placeholder, input, query, cssClasses) {
+  node.type = 'submit';
+  node.title = 'Submit the search query';
+  node.innerHTML = stringNode;
 
-  input.parentNode.appendChild(htmlNode);
+  input.parentNode.appendChild(node);
 }
 
-function addLoadingIndicator(
-  input,
-  loadingIndicator,
-  { loadingIndicator: loadingIndicatorTemplate }
-) {
-  loadingIndicator = {
-    cssClasses: {},
-    template: loadingIndicatorTemplate,
-    ...loadingIndicator,
-  };
-
-  const loadingIndicatorCSSClasses = {
-    root: cx(bem('loading-indicator'), loadingIndicator.cssClasses.root),
-  };
-
+/**
+ * Adds a loading indicator (spinner) to the search box
+ * @param {DomElement} input DOM element where to add the loading indicator
+ * @param {Object} cssClasses css classes definition
+ * @param {Object} templates templates of the widget
+ * @returns {undefined} Modifies the input
+ */
+function addLoadingIndicator(input, cssClasses, templates) {
   const stringNode = renderTemplate({
-    templateKey: 'template',
-    templates: loadingIndicator,
-    data: {
-      cssClasses: loadingIndicatorCSSClasses,
-    },
+    templateKey: 'loadingIndicator',
+    templates,
   });
 
-  const htmlNode = createNodeFromString(
-    stringNode,
-    cx(bem('loading-indicator-wrapper'))
+  const node = document.createElement('span');
+  node.className = cx(
+    suit({ descendantName: 'loadingIndicator' }),
+    cssClasses.loadingIndicator
   );
+  node.innerHTML = stringNode;
 
-  input.parentNode.appendChild(htmlNode);
+  input.parentNode.appendChild(node);
 }
 
-/**
- * Adds a powered by in the searchbox widget
- * @private
- * @param {HTMLElement} input the DOM node of the input of the searchbox
- * @param {object} poweredBy the user options (cssClasses and template)
- * @param {object} templates the default templates
- * @returns {undefined} returns nothing
- */
-function addPoweredBy(input, poweredBy, { poweredBy: poweredbyTemplate }) {
-  // Default values
-  poweredBy = {
-    cssClasses: {},
-    template: poweredbyTemplate,
-    ...poweredBy,
-  };
-
-  const poweredByCSSClasses = {
-    root: cx(bem('powered-by'), poweredBy.cssClasses.root),
-    link: cx(bem('powered-by-link'), poweredBy.cssClasses.link),
-  };
-
-  const url =
-    'https://www.algolia.com/?' +
-    'utm_source=instantsearch.js&' +
-    'utm_medium=website&' +
-    `utm_content=${location.hostname}&` +
-    'utm_campaign=poweredby';
-
-  const stringNode = renderTemplate({
-    templateKey: 'template',
-    templates: poweredBy,
-    data: {
-      cssClasses: poweredByCSSClasses,
-      url,
-    },
-  });
-
-  const htmlNode = createNodeFromString(stringNode);
+function wrapInputFn(input, cssClasses) {
+  const wrapper = document.createElement('div');
+  wrapper.className = cx(suit(), cssClasses.root);
 
-  input.parentNode.insertBefore(htmlNode, input.nextSibling);
-}
+  const form = document.createElement('form');
+  form.className = cx(suit({ descendantName: 'form' }), cssClasses.form);
```

Comment:
is this kept in vanilla JS for combat reasons? 

The form needs an onSubmit handler with preventDefault and the focus moving like in RIS

Reply:
https://blog.algolia.com/mobile-search-ux-tips/ should have most of it explained :) for the rest you can base on the React InstantSearch one, which I know is correct :)

Reply:
Do you think it's good now @Haroenv ? :)

Reply:
I think so based on the code, but the dev novel still has the old code 🤔